streams/src/test/java/org/apache/kafka/streams/integration/RackAwarenessIntegrationTest.java

```diff
@@ -108,55 +117,122 @@ public void cleanup() throws IOException {
         kafkaStreamsInstances.clear();
     }
 
+    @Test
+    public void shouldThrowConfigExceptionWhenRackAwareAssignmentTagsExceedTheLimit() {
```

Comment:
> Maybe also in combination with many tasks.
FYI - current topology generates 6 tasks, I guess that should be enough?

Reply:
Was able to push up to a two hundred stateful tasks and even that was already slow. After 500, test was getting flaky on my computer and sometimes it was just failing due to timeouts.

Feels like these kind of test is too heavy for a local integration tests. Any other way we could do it? Or any pointers to any of the existing integration tests that do the similar thing?

Anyway, I've pushed the changes. I also found one bug in the `ClientTagAwareStandbyTaskAssignor` which is now fixed.

Reply:
I see. I asked some other people and apparently the size of the subscription is not that much of a concern currently. I would still propose to keep the test with 100 instead of 200 tasks. With 200 tasks, the test takes ~10 s on my laptop whereas with 100 tasks it takes ~6 s. I think 6 s is acceptable and might also reduce flakiness. @lkokhreidze and @showuon WDYT?

Reply:
Sounds good to me. Will push changes with 100 stateful tasks for the rebalancing test.

Reply:
Pushed changes to reduce number of stateful tasks to 100
https://github.com/apache/kafka/pull/11837/commits/e6f717229acb4850aec1f33e13cd88cdc3ccb4bb

Reply:
Even with 100 tasks tests on CI failed with timeout. I've bumped timeout to 1.5 minutes in the latest commit.

Reply:
If after your PR the tests are still flaky, let's just test with 6 tasks and merge this PR. You can then add a system test for the scenario with larger subscriptions in a follow-up PR.

Reply:
Thanks @cadonna 
If no objections, I will do that now. The fact that it times out in 1 minute doesn't give much confidence that it won't be flaky.
I will follow up the system test.
fyi @showuon